src/main/java/heronarts/lx/midi/surface/APC40Mk2.java

```diff
@@ -370,9 +456,46 @@ void onDeviceOnOff() {
     }
 
     void onKnob(int index, double normalized) {
-      if (this.knobs[index] != null) {
-        this.knobs[index].setNormalized(normalized);
+      LXListenableParameter knob = this.knobs[index];
+
+      if (knob == null) {
+        return;
+      }
+
+      // If the knob's an LCP, touching it should link it to the focusColor,
+      // if one is set and it belongs to the main palette and not a side
+      // swatch, or set it static and initialize it to the colorClipboard,
+      // if there is one. Otherwise, have it control the value as normal.
+      if (knob instanceof LinkedColorParameter) {
+        LinkedColorParameter lcp = (LinkedColorParameter) knob;
+
+        if (focusColor != null) {
+          int palIndex = lx.engine.palette.swatch.colors.indexOf(focusColor);
+          if (palIndex >= 0) {
+            lcp.mode.setValue(LinkedColorParameter.Mode.PALETTE);
+            lcp.index.setValue(palIndex + 1);
+            return;
+          }
+        }
+        if (colorClipboard != null) {
```

Comment:
Similarly here, I think this part should work for a basic `ColorParameter` as well, where the color value can be copied without linking.

Reply:
Done